doc/reference/tooling/tt_cli/cluster.rst

```diff
@@ -1,10 +1,242 @@
-.. _tt-cluster:
+.. _tt-cluster2:
 
-Managing a cluster's configuration
-==================================
+Managing cluster configurations
+===============================
 
-..  code-block:: console
+.. code-block:: console
 
-    $ tt cluster
+    $ tt cluster COMMAND {APPLICATION[:APP_INSTANCE] | URI} [FILE] [OPTION ...]
 
-.. TODO: https://github.com/tarantool/doc/issues/3725
+``tt cluster`` manages :ref:`YAML configurations <configuration>` of Tarantool applications.
+This command works both with local configuration files in application directories
+and with centralized configuration storages (:ref:`etcd <configuration_etcd>` or Tarantool-based).
+
+``COMMAND`` is one of the following:
+
+*   ``publish``: publish a cluster configuration from a file.
+*   ``show``: print a cluster configuration.
+
+.. _tt-cluster-local:
+
+Managing local configurations
+-----------------------------
+
+``tt cluster`` can read and modify local cluster configurations stored in the
+``config.yaml`` files inside application directories.
+
+To write a configuration to a local ``config.yaml``, run ``tt cluster publish``
+with two arguments:
+
+*   the application name.
+*   the path to a YAML file from which the configuration should be taken.
+
+.. code-block:: console
+
+    $ tt cluster publish myapp source.yaml
+
+To print a local configuration from ``config.yaml`` in an application directory,
+run ``tt cluster show`` with the application name:
+
+.. code-block:: console
+
+    $ tt cluster show myapp
+
+.. _tt-cluster-centralized:
+
+Managing configurations in centralized storages
+-----------------------------------------------
+
+``tt cluster`` can manage centralized cluster configurations in storages of both
+supported types: :ref:`etcd <configuration_etcd>` or a Tarantool-based configuration storage.
+
+To send a configuration from a file to a centralized configuration storage,
+run ``tt cluster publish`` with a URI of this storage's
+instance as the target. For example, to send a configuration from ``source.yaml``
+to a local etcd instance on the default port ``2379``:
+
+.. code-block:: console
+
+    $ tt cluster publish "http://localhost:2379/myapp" source.yaml
+
+A URI must include a prefix that is unique for the application. It can also include
+credentials and other connection parameters. Find the detailed description of the
+URI format in :ref:`tt-cluster-centralized-uri`.
+
+To print a cluster configuration from a centralized storage, run ``tt cluster show``
+with a storage URI including the prefix identifying the application. For example, to print
+``myapp``'s configuration from a local etcd storage:
+
+.. code-block:: console
+
+    $ tt cluster show "http://localhost:2379/myapp"
+
+.. _tt-cluster-centralized-authentication:
+
+Authentication
+~~~~~~~~~~~~~~
+
+There are three ways to pass the credentials for connecting to the centralized configuration storage:
```

Comment:
Looks like they apply to etcd only? Or `Credentials specified in the storage URI` applies to Tarantool-config storage, too?

Reply:
All three methods apply to both storage types. Even env variables that have ETCD in their names...

Reply:
> Even env variables that have ETCD in their names...

OMG!